### PR TITLE
05 cz narrative

### PR DESCRIPTION
**Pull request by Carlson-Lee**
Issue: #94

### CODE REVIEW

**Comment by Carlson-Lee** [comment]
A little heads up with the triple pop-ups with the buildings, central buildings and bonus tiles. I was trying to trigger them individually, but I couldn't get that done properly.
image

File: Assets/Scripts/TutorialManager.cs [hunk 1]
```diff
@@ -38,10 +47,22 @@ public class TutorialManager : MonoBehaviour
...
+    
 
-    // Start is called before the first frame update
-
+    private Dictionary<TutorialSections, string> narrativeTexts = new Dictionary<TutorialSections, string>()
+    {
+        { TutorialSections.Building, "Let's start by learning how to build. Place buildings on the highlighted tiles." },
+        { TutorialSections.BonusTiles, "The coloured space on the tiles you see, they are bonus tiles, place an building on it for extra scores." },
+        { TutorialSections.Wiring, "Now, let's connect the building to the central building using wiring." },
+        { TutorialSections.Scoreboard, "Congratulation you have score a point. - Click to Advance" },
```
Comment (PBWiseman) [inline Assets/Scripts/TutorialManager.cs:58]: Should be "Congratulations you have scored a point."

File: Assets/Scripts/TutorialManager.cs [hunk 2]
```diff
@@ -38,10 +47,22 @@ public class TutorialManager : MonoBehaviour
...
     private GameObject exitBTN;
+    [SerializeField]
+    
 
-    // Start is called before the first frame update
-
+    private Dictionary<TutorialSections, string> narrativeTexts = new Dictionary<TutorialSections, string>()
+    {
+        { TutorialSections.Building, "Let's start by learning how to build. Place buildings on the highlighted tiles." },
+        { TutorialSections.BonusTiles, "The coloured space on the tiles you see, they are bonus tiles, place an building on it for extra scores." },
```
Comment (PBWiseman) [inline Assets/Scripts/TutorialManager.cs:56]: Should be "a building"

**Review by PBWiseman — CHANGES_REQUESTED** [review]
I like the rework! Couple of minor spelling mistakes to fix then it should be good to merge. The only other thing is that there seems to be conflicting changes with Level1 scene? Not totally sure why since all your work seems to be in the Tutorial scene. Just make sure that you keep the staging changes when merging.

**Review by PBWiseman — APPROVED** [review]

**Review by LiamMcKenzie — CHANGES_REQUESTED** [review]
the rocks at the bottom don't have any collision, this can be fixed in TutorialManager.cs

Otherwise I think this looks good

**Review by ChaseBENNC9 — CHANGES_REQUESTED** [review]
New Tutorial feels much better with the added narrative. well done. As mentioned already there is an issue with the obstacle's collision

**Review by ChaseBENNC9 — APPROVED** [review]
looks good

**Review by LiamMcKenzie — APPROVED** [review]
# Code Review

## Functional Requirements
_Is the application fit for purpose? Does it meet the user’s requirements and properly address the issue?_  

 - I think this is an improvement to the old tutorial
 - The changes I requested were addressed

## Legibility and Style
_Uses agreed coding standards/conventions? Readable, consistent and well-structured with useful comments?_  

 - It could have used more comments, but there are enough to understand what the bigger functions are doing.

_Any repetitive code that could be refactored? (DRY)_  

 - No repeated code from what I noticed. 
 - The code to say where buildings is looks a bit repetitive, but we've all agreed this is okay.

## Correctness of the Code
_Is the code reliable, giving same results every time it runs? Does it function correctly?_  

 - No bugs from what I noticed

## Implementation
_Any notable performance/speed issues? Compatibility and Responsiveness on other platforms ?_  

 - I noticed no speed issues.


# Final Comments

 - This is a great feature, the updated text is much easier to understand.
 - Looks good to merge into staging

***

**Comment by LiamMcKenzie** [comment]
Merged on behalf of carlson

**Pull request merged**